### PR TITLE
Register UncreatableType Explcitily

### PR DESCRIPTION
**Pull request by rakeshsadhu**

### CODE REVIEW

File: src/model/QMLFrontend.h [hunk 1]
```diff
@@ -183,6 +183,16 @@ QObject *singletonGetterByFunction(QQmlEngine *engine, QJSEngine *scriptEngine)
 }
 
 
+template<typename ProviderType>
+int registerUncreatableQmlComponent(const char *uri, const char *name = ProviderType::INTERFACE_NAME, int majorVersion =
```
[the remark this answers is not in the document]
Comment (rakeshsadhu) [reply src/model/QMLFrontend.h]: why ? is not it  that we want to registeruncreatetype ?

File: src/model/QMLFrontend.h [hunk 2]
```diff
@@ -183,6 +183,16 @@ QObject *singletonGetterByFunction(QQmlEngine *engine, QJSEngine *scriptEngine)
 }
 
 
+template<typename ProviderType>
```
[the remark this answers is not in the document]
Comment (rakeshsadhu) [reply src/model/QMLFrontend.h]: why not?

File: examples/AddressBook/AddressBookPlugin.cpp [hunk 3]
```diff
@@ -43,7 +43,8 @@ void AddressBookPlugin::registerTypes(const char *uri)
 {
     // Register the generated types
     Module::registerQmlTypes(uri);
-    Module::registerUncreatableQmlTypes(uri);
+
+    facelift::registerUncreatableQmlComponent<AddressBookCppWithProperties>(uri, "AddressBookUncreateAble");
```
Comment (jacky309) [inline examples/AddressBook/AddressBookPlugin.cpp]: We don't want to use "AddressBookUncreateAble" in any QML code.
"AddressBook"

File: src/model/QMLFrontend.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (jacky309) [reply src/model/QMLFrontend.h]: We want the interface to be registered as uncreatable type

**Pull request closed without merging**